.github/workflows/ci-e2e-badger.yaml

```diff
@@ -40,6 +40,22 @@ jobs:
             STORAGE=badger make jaeger-v2-storage-integration-test
             ;;
         esac
+          
+    - name: Upload Metrics Artifact
+      uses: actions/upload-artifact@v4
+      if: ${{ matrix.version }} == "v2"
+      with:
+        name: badger-metrics-${{ matrix.version }}.txt
+        path: ./.metrics/badger_metrics.txt
+        retention-days: 7
+  
+    - name: Release Action then cache 
+      uses: actions/cache@v4
+      if: startsWith(github.ref, 'refs/tags/')  && ${{ matrix.version }} == "v2"
+      with:
+        path: ./.metrics/badger_metrics.txt
+        key: ${{ github.ref }}_badger_metrics
```

Comment:
q: if you are using github.ref that has slashes in them, what would it do to the artifact file name when it's time to download it?

Reply:
i thinking of uploading it with tags_{file_name} during a tagged workflow run and using another github action https://github.com/actions-ecosystem/action-get-latest-tag we can fetch the lastest tag and create the key to fetch the latest cache